### PR TITLE
Add content disposition header in resource proxy endpoint

### PR DESCRIPTION
**Pull request by takirala**
Add content disposition header to make sure the proxied filename is consistent. Addresses [DCOS_OSS-1802](https://jira.mesosphere.com/browse/DCOS_OSS-1802)

### CODE REVIEW

**Comment by takirala** [comment]
@jsancio I used a dcos cluster that has this PR version of cosmos (with help of https://github.com/dcos/dcos/pull/2049) and used the local dcos cli (with help of https://github.com/dcos/dcos-cli/pull/1090) and was able to resolve the cassandra installation bug.

File: cosmos-test-common/src/test/scala/com/mesosphere/cosmos/handler/ResourceProxyHandlerSpec.scala [hunk 1]
```diff
@@ -153,6 +154,36 @@ final class ResourceProxyHandlerSpec extends FreeSpec with PropertyChecks {
...
+
+    // These should never happen, but just in case.
+
+    ResourceProxyHandler.getFileNameFromUrl(
+      Uri.parse("http://doesntreallymatter.com/")
+    ) shouldEqual "doesntreallymatter.com"
+
+    ResourceProxyHandler.getFileNameFromUrl(
+      Uri.parse("https://doesntreallymatter.com")
+    ) shouldEqual "doesntreallymatter.com"
```
Comment (jsancio) [inline cosmos-test-common/src/test/scala/com/mesosphere/cosmos/handler/ResourceProxyHandlerSpec.scala]: Hmm. Interesting. I say we shouldn't set the content disposition in this case. Meaning `getFileNameFromUrl` should return a `Option[String]` and we should only set the content disposition when it is `Some`.

**Comment by jsancio** [comment]
@takirala Can you record a demo of the CLI using this feature? I am interested to see how the CLI handles the content-disposition header. You can test this by installing Cassandra (`dcos package install cassandra`) and try run a command for the CLI (`dcos cassandra ...`).

**Comment by takirala** [comment]
@jsancio 
The cassandra cli is working as expected image

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/handler/ResourceProxyHandler.scala [hunk 2]
```diff
@@ -43,6 +44,11 @@ final class ResourceProxyHandler private(
           response.content = Buf.ByteArray.Owned(contentBytes)
           response.contentType = responseData.contentType.show
           response.contentLength = contentBytes.length.toLong
+          getFileNameFromUrl(uri) match {
+            case Some(filename) => response.headerMap.add(Fields.ContentDisposition,
+              s"""attachment; filename="$filename"""")
+            case _ =>
+          }
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/handler/ResourceProxyHandler.scala]: This is `foreach`:

```
getFileNameFromUrl(uri).foreach { filename =>
  ...
}
```
Or using Scala's syntactic sugar:
```
for (filename <- getFileNameFromUrl(uri)) {
  ...
}
```

**Pull request merged**